Student-Unity/src/main/java/csed/swe/studentunity/Logic/AllCourseService.java

```diff
@@ -64,22 +68,36 @@ public List<Course> getAllActiveCourses(){
         return courseRepo.findCourseByStatus(true);
     }
 
-    public ResponseEntity<Set<Course>> getRegisteredCourses(String sessionId){
+    public ResponseEntity<List<?>> getRegisteredCourses(String sessionId){
         ActiveUserService activeUserService = ActiveUserService.getInstance();
         String userEmail = activeUserService.checkLogin(UUID.fromString(sessionId))[0];
         User user = userService.getUser(userEmail).orElse(null);
         if (user != null)
-            return new ResponseEntity<>(new HashSet<>(), HttpStatus.OK);
-        return new ResponseEntity<>(new HashSet<>(), HttpStatus.NOT_FOUND);
+            return new ResponseEntity<>(registeredCourseRepository.getRegisteredCourseByUserId(user.getId()), HttpStatus.OK);
+        return new ResponseEntity<>(new ArrayList<>(), HttpStatus.NOT_FOUND);
     }
 
-    public HttpStatusCode registerCourse(String sessionId, String courseCode) {
+    public HttpStatusCode registerCourse(String sessionId, long courseId) {
         ActiveUserService activeUserService = ActiveUserService.getInstance();
         String userEmail = activeUserService.checkLogin(UUID.fromString(sessionId))[0];
         User user = userService.getUser(userEmail).orElse(null);
-        Course course = courseRepo.findCourseByCode(courseCode).orElse(null);
-        // if (user != null && course != null)
-            // Todo:
+        Course course = courseRepo.findCourseById(courseId).orElse(null);
+        if (user != null && course != null) {
+            registeredCourseRepository.save(new RegisteredCourse(courseId, user.getId(), false));
+            return HttpStatus.OK;
+        }
+        return HttpStatus.NOT_FOUND;
+    }
+
+    public HttpStatusCode unRegisterCourse(String sessionId, long courseId) {
+        ActiveUserService activeUserService = ActiveUserService.getInstance();
+        String userEmail = activeUserService.checkLogin(UUID.fromString(sessionId))[0];
+        User user = userService.getUser(userEmail).orElse(null);
+        Course course = courseRepo.findCourseById(courseId).orElse(null);
+        if (user != null && course != null) {
+            registeredCourseRepository.deleteRegisteredCourseById(user.getId(), courseId);
+            return HttpStatus.OK;
```

Comment:
The Http status are good just do them in the API layer and we will check for them in the front end 
possible solution you can return Boolean from this function